cmgd/cmgd_db.h

```diff
@@ -56,6 +56,18 @@ typedef void (*cmgd_db_node_iter_fn)(cmgd_db_hndl_t db_hndl,
 
 extern const char *cmgd_db_names[CMGD_DB_MAX_ID+1];
 
+typedef struct cmgd_db_ctxt_ {
```

Comment:
No please return this back to cmgd_db.c. Move all commit info stuff to cmd_db.c instead. There's no need to expose commit-infos to rest of the other modules. Why do you need to expose cmd_db_ctxt_t to rest of the other modules. Please think modular while designing/coding.